src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -49,45 +61,123 @@ struct KeyValueViewModel {
-            } else {
-                // Convert into String in generated code
-                self.value = "String(\(param.name))"
+            self.constantValue = nil
+
+            // value is referring a signature parameter, no need to wrap as String
+            self.value = convertValueToStringInSwift(
+                type: signatureParameter.schema.type,
+                value: name
+            )
+
+            // if parameter is from method signature (not from option) and type is date or byteArray,
+            // add decoding logic to string in the method and specify the right decoding strategy
+            switch signatureParameter.schema.type {
+            case .date,
+                 .unixTime,
+                 .dateTime:
+                keyValueType = signatureParameter.required ? .dateFromSignature : .dateFromConstant
+                self.needDecodingInMethod = true
+            case .byteArray:
+                keyValueType = signatureParameter.required ? .byteArrayFromSignature : .byteArrayFromConstant
+                self.needDecodingInMethod = true
+            default:
+                self.needDecodingInMethod = false
             }
+        } else if param.implementation == ImplementationLocation.client {
+            self.value = "client.\(name)"
+            self.optional = false
+            self.constantValue = nil
+            self.needDecodingInMethod = false
         } else {
             self.value = ""
             self.optional = false
-            self.paramName = nil
+            self.constantValue = nil
+            self.needDecodingInMethod = false
         }
+
+        self.key = name
+        self.strategy = keyValueType.rawValue
     }
 
     /**
         Create a ViewModel with a Key and Value pair
-
         - Parameter key: Key String in the Key value pair
         - Parameter value: the value string
      */
     init(key: String, value: String) {
         self.key = key
         self.value = value
         self.optional = false
-        self.paramName = nil
+
+        self.strategy = KeyValueDecodeStrategy.default.rawValue
+        self.needDecodingInMethod = false
+        self.constantValue = nil
+    }
+}
+
+/**
+ Convert the type into String format in Swift
+ */
+private func convertValueToStringInSwift(type: AllSchemaTypes, value: String, key: String? = nil) -> String {
+    switch type {
+    case .string:
+        return "\(key ?? value)"
+    case .integer,
+         .number,
+         .boolean:
+        return "String(\(key ?? value))"
+    // For these types, a variable will be created in the method using the naming convention `{key|value}String`
+    case .date,
+         .unixTime,
+         .dateTime,
+         .byteArray:
+        return "\(key ?? value)String"
+    case .choice,
+         .sealedChoice:
+        return "\(value).rawValue"
+    case .array:
+        return "\(value).map { String($0) }.joined(separator: \",\") "
+    default:
+        return "\(value)"
     }
 }
```

Comment:
I think this would be unnecessary if `ConstantValue.value` where declared type `Any`

Reply:
Rename the function to clarify the purpose of this function.